numpy/core/numeric.py

```diff
@@ -1457,6 +1461,93 @@ def rollaxis(a, axis, start=0):
     return a.transpose(axes)
 
 
+def _validate_axis(axis, ndim, argname):
+    try:
+        axis = [operator.index(axis)]
+    except TypeError:
+        axis = list(axis)
+    original_axis = axis
+    axis = [a + ndim if a < 0 else a for a in axis]
+    if not builtins.all(0 <= a < ndim for a in axis):
+        raise ValueError('%s=%s has elements outside valid bounds' %
```

Comment:
transpose uses ValueError: invalid axis for this array. I like printing the argname and the passed in axes, but the wording seems a bit more clear than valid bounds.